ed25519_utils.go

```diff
@@ -0,0 +1,29 @@
+package jwt
+
+import (
+	"encoding/pem"
+	"golang.org/x/crypto/ed25519"
+)
+
+
+// Parse PEM encoded Elliptic Curve Private Key Structure
+func ParseED25519PrivateKeyFromPEM(key []byte) (*ed25519.PrivateKey, error) {
+	var block *pem.Block
+	if block, _ = pem.Decode(key); block == nil {
+		return nil, ErrKeyMustBePEMEncoded
+	}
+
+	privKey := ed25519.NewKeyFromSeed(block.Bytes[16:])
+	return &privKey, nil
+}
+
+// Parse PEM encoded PKCS1 or PKCS8 public key
+func ParseED25519PublicKeyFromPEM(key []byte) (*ed25519.PublicKey, error) {
+	var block *pem.Block
+	if block, _ = pem.Decode(key); block == nil {
```

Comment:
This is probably wrong. EC public keys contain:

```
0x30819b // 0x9b bytes
0x3010 [10 bytes] // ?
0x038186 [134 bytes] // public key
```

Though I'm not really sure what this would look like for Ed25519. I think the Parse* functions should just be dropped.

Reply:
It might really be complete nonsense but it produced correct results for multiple keys but that could've been just coincidence.